via/services/url_details.py

```diff
@@ -21,19 +28,19 @@ def get_url_details(self, url, headers=None):
         :return: 2-tuple of (mime type, status code)
 
         :raise BadURL: When the URL is malformed
+        :raise checkmatelib.BadURL: if the URL is blocked by Checkmate
         :raise UpstreamServiceError: If we server gives us errors
         :raise UnhandledException: For all other request based errors
         """
-        if headers is None:
-            headers = OrderedDict()
+        if self._youtube.enabled and self._youtube.get_video_id(url):
+            return "video/x-youtube", 200
```

Comment:
Moved the YouTube check up to the top of the method because this is the one thing that we want to do _before_ calling Checkmate.

Inside this YouTube `if` is where we could do a _different_ call to Checkmate instead, with different ignore reasons, if we wanted to still allow some YouTube videos to be specifically blocked. But that would also require some changes in other files, so that's for a separate PR.